src/coord/timestamp.rs

```diff
@@ -244,6 +340,109 @@ pub struct Timestamper {
     max_increment_size: i64,
 }
 
+/// A debezium record contains a set of update counts for each topic that the transaction
+/// updated. This function extracts the set of (topic, update_count) as a vector.
+fn parse_debezium(record: Vec<(String, Value)>) -> Vec<(String, i64)> {
+    let mut result = vec![];
+    for (key, value) in record {
+        if key == "data_collections" {
+            if let Value::Union(_, value) = value {
+                if let Value::Array(items) = *value {
+                    for v in items {
+                        if let Value::Record(item) = v {
+                            let mut value: String = String::new();
+                            let mut write_count = 0;
+                            for (k, v) in item {
+                                if k == "data_collection" {
+                                    if let Value::String(data) = v {
+                                        value = data;
+                                    } else {
+                                        error!("Incorrect AVRO format. String expected");
```

Comment:
You can (and should) just panic here. We know the record matched schema and so if the record decoded successfully, either this is a bug in the Avro parser or a bug in this function. Either way it deserves to be much louder than a log error message—that's likely to silently break consistency unless you're staring at your log files, I think?